Source/OxyPlot.Avalonia.Shared/PlotBase.cs

```diff
@@ -0,0 +1,250 @@
+        protected Panel panel;
+        private Canvas overlays;
+        private ContentControl zoomControl;
+        private ScreenPoint mouseDownPoint;
+
+        protected const string PartPanel = "PART_Panel";
+
+        /// <summary>
+        /// Invalidation flag (0: no update, 1: update, 2: update date).
+        /// </summary>
+        protected internal int isUpdateRequired;
+
+        /// <summary>
+        /// Invalidation flag (0: no update, 1: update visual elements).
+        /// </summary>
+        protected internal int isRenderRequired;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PlotBase" /> class.
+        /// </summary>
+        protected PlotBase()
+        {
+            this.TrackerDefinitions = [];
+            this.GetObservable(BoundsProperty).Subscribe(new AnonymousObserver<Rect>(this.OnSizeChanged));
+        }
+
+        /// <summary>
+        /// Gets the coordinates of the client area of the view.
+        /// </summary>
+        public OxyRect ClientArea => new(0, 0, this.Bounds.Width, this.Bounds.Height);
+
+        /// <summary>
+        /// Gets the tracker definitions.
+        /// </summary>
+        /// <value>The tracker definitions.</value>
+        public ObservableCollection<TrackerDefinition> TrackerDefinitions { get; }
+
+        /// <summary>
+        /// Hides the tracker.
+        /// </summary>
+        public void HideTracker()
+        {
+            if (this.currentTracker != null)
+            {
+                this.overlays.Children.Remove(this.currentTracker);
+                this.currentTracker = null;
+            }
+        }
+
+        /// <summary>
+        /// Hides the zoom rectangle.
+        /// </summary>
+        public void HideZoomRectangle()
+        {
+            this.zoomControl.IsVisible = false;
+        }
+
+        /// <summary>
+        /// Pans all axes.
+        /// </summary>
+        /// <param name="delta">The delta.</param>
+        public void PanAllAxes(Vector delta)
+        {
+            this.ActualModel?.PanAllAxes(delta.X, delta.Y);
+            this.InvalidatePlot(false);
+        }
+
+        /// <summary>
+        /// Zooms all axes.
+        /// </summary>
+        /// <param name="factor">The zoom factor.</param>
+        public void ZoomAllAxes(double factor)
+        {
+            this.ActualModel?.ZoomAllAxes(factor);
+            this.InvalidatePlot(false);
+        }
+
+        /// <summary>
+        /// Resets all axes.
+        /// </summary>
+        public void ResetAllAxes()
+        {
+            this.ActualModel?.ResetAllAxes();
+            this.InvalidatePlot(false);
+        }
+
+        /// <summary>
+        /// Invalidate the PlotView (not blocking the UI thread)
+        /// </summary>
+        /// <param name="updateData">The update Data.</param>
+        public virtual void InvalidatePlot(bool updateData = true)
+        {
+            this.isUpdateRequired = updateData ? 2 : 1;
```

Comment:
This isn't sufficient: may change a 2 to a 1

Reply:
https://github.com/oxyplot/oxyplot-avalonia/blob/9913a97d79a75585abd0ee5b417739ac52bca4c6/Source/OxyPlot.Avalonia/PlotBase.cs#L232
Old code's purpose is to ensure we only ever increase the value (so we can't go from requesting an update with data-changes to an update without data-changes) and only fires off a request if there isn't one active already (i.e. if the user calls `plot.InvalidatePlot(true)` followed immediately by `plot.InvalidatePlot(false)`, the old code will ensure that if the work isn't picked up by the thread doing the update before the second call that it will still observe the `updateData=true` bit and it will only be invoked once)

Reply:
Updated the code to make sure a 2 is not overwritten by a 1. Regarding the rest of the old behaviour (only one invoke for consecutive update calls), I will have to look into it some more later.